File: scripts/convert-to-webp.py
"""
FRIKAT AUTO — Convert images to WebP
Scans images/, converts jpg/png to webp, updates _vehicules/*.md, deletes originals.
"""

import os
import re
from PIL import Image

IMAGES_DIR   = os.path.join(os.path.dirname(__file__), '..', 'images')
VEHICULES_DIR = os.path.join(os.path.dirname(__file__), '..', '_vehicules')
CONVERTIBLE  = {'.jpg', '.jpeg', '.png'}

def run():
    files = [f for f in os.listdir(IMAGES_DIR)
             if os.path.splitext(f)[1].lower() in CONVERTIBLE]

    if not files:
        print('Nothing to convert.')
        return

    renamed = {}

    for f in files:
        src  = os.path.join(IMAGES_DIR, f)
        base = os.path.splitext(f)[0]
        dst_name = base + '.webp'
        dst  = os.path.join(IMAGES_DIR, dst_name)

        before = os.path.getsize(src) // 1024

        from PIL import ImageOps
        img = ImageOps.exif_transpose(Image.open(src)).convert('RGB')
        img.save(dst, 'WEBP', quality=85)
        os.remove(src)

        after = os.path.getsize(dst) // 1024
        saved = round((1 - after / max(before, 1)) * 100)
        renamed[f'images/{f}'] = f'images/{dst_name}'
        print(f'  ✓  {f} → {dst_name}  ({before}KB → {after}KB, -{saved}%)')

    if not renamed:
        return

    for md_file in os.listdir(VEHICULES_DIR):
        if not md_file.endswith('.md'):
            continue
        md_path = os.path.join(VEHICULES_DIR, md_file)
        content = open(md_path, encoding='utf-8').read()
        new_content = content
        for old, new in renamed.items():
            new_content = new_content.replace(old, new)
        if new_content != content:
            open(md_path, 'w', encoding='utf-8').write(new_content)
            print(f'  ↺  Updated: _vehicules/{md_file}')

    print(f'\nDone. {len(renamed)} image(s) converted.')

if __name__ == '__main__':
    run()

File: scripts/convert-to-webp.js
/**
 * FRIKAT AUTO — Conversion images → WebP
 * Usage: node scripts/convert-to-webp.js
 *
 * - Scans images/ for jpg/jpeg/png files
 * - Converts each to .webp (quality 85)
 * - Updates all _vehicules/*.md references
 * - Deletes the original files
 */

const sharp  = require('sharp');
const fs     = require('fs');
const path   = require('path');

const IMAGES_DIR   = path.join(__dirname, '..', 'images');
const VEHICULES_DIR = path.join(__dirname, '..', '_vehicules');
const CONVERTIBLE  = ['.jpg', '.jpeg', '.png'];

async function run() {
    // 1. Find all non-webp images
    const files = fs.readdirSync(IMAGES_DIR).filter(f => {
        const ext = path.extname(f).toLowerCase();
        return CONVERTIBLE.includes(ext);
    });

    if (files.length === 0) {
        console.log('✓ No images to convert — all already .webp');
        return;
    }

    console.log(`Found ${files.length} image(s) to convert:\n`);

    const renamed = {}; // { 'images/old.jpg': 'images/new.webp' }

    // 2. Convert each image
    for (const file of files) {
        const srcPath  = path.join(IMAGES_DIR, file);
        const baseName = path.basename(file, path.extname(file));
        const newFile  = baseName + '.webp';
        const dstPath  = path.join(IMAGES_DIR, newFile);

        // Skip if webp already exists with same name
        if (fs.existsSync(dstPath)) {
            console.log(`  SKIP  ${file} → ${newFile} (already exists)`);
            renamed[`images/${file}`] = `images/${newFile}`;
            fs.unlinkSync(srcPath);
            continue;
        }

        try {
            const before = fs.statSync(srcPath).size;
            await sharp(srcPath).webp({ quality: 85 }).toFile(dstPath);
            const after = fs.statSync(dstPath).size;
            const saved = Math.round((1 - after / before) * 100);

            fs.unlinkSync(srcPath);

            renamed[`images/${file}`] = `images/${newFile}`;
            console.log(`  ✓  ${file} → ${newFile}  (${Math.round(before/1024)}KB → ${Math.round(after/1024)}KB, -${saved}%)`);
        } catch (err) {
            console.error(`  ✗  ${file} — Error: ${err.message}`);
        }
    }

    // 3. Update .md files
    if (Object.keys(renamed).length === 0) return;

    const mdFiles = fs.readdirSync(VEHICULES_DIR).filter(f => f.endsWith('.md'));
    let mdUpdated = 0;

    for (const mdFile of mdFiles) {
        const mdPath = path.join(VEHICULES_DIR, mdFile);
        let content  = fs.readFileSync(mdPath, 'utf8');
        let changed  = false;

        for (const [oldPath, newPath] of Object.entries(renamed)) {
            if (content.includes(oldPath)) {
                content = content.split(oldPath).join(newPath);
                changed = true;
            }
        }

        if (changed) {
            fs.writeFileSync(mdPath, content, 'utf8');
            console.log(`  ↺  Updated: _vehicules/${mdFile}`);
            mdUpdated++;
        }
    }

    console.log(`\nDone. ${Object.keys(renamed).length} image(s) converted, ${mdUpdated} .md file(s) updated.`);
    console.log('→ Run: git add -A && git commit -m "Convert images to webp"');
}

run().catch(err => {
    console.error('Fatal error:', err.message);
    process.exit(1);
});
